hkmc2/shared/src/main/scala/hkmc2/syntax/Lexer.scala

```diff
@@ -397,7 +397,15 @@ class Lexer(origin: Origin, dbg: Bool)(using raise: Raise):
               // *     1
               // * which results in token stream `|module| |P| |with|→|(|→|2|)|←|↵|1|`.
               // * So this code commutes the indent/deindent with the open/close parentheses.
-              go(CLOSE_BRACKET(Indent) -> l1.left :: (CLOSE_BRACKET(k1), l1) :: OPEN_BRACKET(Indent) -> l1.right :: rest, false, stack, acc)
+              rest match
+                case (OPEN_BRACKET(k2), l0) :: rest if k1 == Round && k2 == Round =>
+                  // * Special case for immediate lambda applications; e.g.,
+                  // * (x =>
+                  // *   x + 1)(0)
+                  // * do not insert indents around `(0)`
+                  go(CLOSE_BRACKET(Indent) -> l1.left :: (CLOSE_BRACKET(k1), l1) :: (OPEN_BRACKET(k1), l1) :: rest, false, stack, acc)
```

Comment:
This is a hack. Please remove it and either fix the real problem at its source or open an issue for it.

Reply:
This problem looks a bit tricky. I've opened an issue (#281) for it and left a `fixme` test case.